### ISSUE TITLE
Enum reserved word even for php<8.1

### ISSUE THREAD

**Issue by MasonD** [prettier/plugin-php#1942]
**PHP Plugin 0.18.2**

**Config:**
```yml
phpVersion: '7.4'
```
OR any other version of php below 8.1, such as 8.0

**Input:**

```php
<?php
use Name\Space\Enum;
```
OR
```php
<?php
class Enum {}
```

**Output:**

```
Parse Error : syntax error, unexpected 'Enum' (T_ENUM), expecting ';'
```
OR
```
Parse Error : syntax error, unexpected 'Enum' (T_ENUM), expecting T_STRING
```

**Expected behavior:**
No parse error as Enum isn't a reserved word in php <8.1.

I understand that this might be an error with `php-parser`, but I thought it might also be plugin-php not telling php-parser the php version correctly, and I don't have an environment set up to test which is the case, so I reported it here because I'm experiencing it when using plugin-php.

**Comment by czosel** [comment]
Thanks for the bug report @MasonD. First I assumed it's an issue in the plugin, because indeed we are not passing `phpVersion` further to the parser currently. But it turns out that the parser is not able to handle `class Enum {}` even for PHP versions before 8.1. I created an issue in the parser repo: https://github.com/glayzzle/php-parser/issues/896

**Comment by Berbass** [comment]
Having the same issue here...
Any news on it?

**Comment by czosel** [comment]
Fixed in `v0.18.5` :tada:

**Comment by djschilling** [comment]
I just tested with `v0.18.5` and still get this error:

```
[error] system/src/Lib/SocialMedia/SocialMedia.php: SyntaxError: Parse Error : syntax error, unexpected 'Enum' (T_ENUM), expecting T_STRING on line 22
[error]   20 |  * @psalm-immutable
[error]   21 |  */
[error] > 22 | class SocialMedia extends Enum {
[error]      |                          ^
[error]   23 |     private const FACEBOOK = 'Facebook';
[error]   24 |     private const INSTAGRAM = 'Instagram';
[error]   25 |     private const PINTEREST = 'Pinterest';
```

**Comment by czosel** [comment]
Thanks for the feedback @djschilling - the latest release only fixed the issue with keywords in namespaces, but not in class names. I'll reopen this issue.

**Comment by czosel** [comment]
Fixed in `v0.18.7` :tada:
